### DISCUSSION TITLE
Reading order reconstruction

### DISCUSSION THREAD

**Discussion by maiiabocharova** [PaddlePaddle/PaddleOCR#7699] [Q&A] [answered]
Hello, in the Readme of KIE it is mentioned that the algorithm first sorts the words to reconstruct the reading order and than passes the sorted words to VI Layoutxml / Layoutxml.

Can yiu please point me to this sorting algorithm?

Thank you

**Comment by littletomatodonkey** [comment, marked as answer]
Hi, the detailed implementation is here: https://github.com/PaddlePaddle/PaddleOCR/blob/76fefd56f86f2809a6e8f719220746442fbab9f3/ppocr/data/imaug/vqa/augment.py#L22

The effect is as follows.

![8c0194536eda5942a41f6a266932f2e2](https://user-images.githubusercontent.com/14270174/192766246-329b1ed1-af98-43dc-81da-9d304cbc40f0.jpg)

We will also open-source our tech-report. Thank you for your attention.

**Reply by maiiabocharova** [reply]
Thank you a lot!

**Comment by Nkosikhona-Ndlovu** [comment]
Hello @maiiabocharova and @littletomatodonkey.
Within the function order_by_tbyx(ocr_info): is what I assume to be the threshold at 20 pixels for the TB-YX sorting algorithm. Is the threshold to be hard coded into this function or can it be specified dynamically elsewhere within the paddleOCR framework? 

Thanks

**Reply by reBiocoder** [reply]
Whether the threshold can be dynamically adjusted？
